website/src/components/SearchPage/DownloadDialog/FieldSelector/FieldSelectorModal.tsx

```diff
@@ -0,0 +1,171 @@
+import { useState, type FC } from 'react';
+
+import { ACCESSION_VERSION_FIELD } from '../../../../settings.ts';
+import { type Metadata } from '../../../../types/config.ts';
+import { BaseDialog } from '../../../common/BaseDialog.tsx';
+
+type FieldSelectorProps = {
+    isOpen: boolean;
+    onClose: () => void;
+    metadata: Metadata[];
+    initialSelectedFields?: string[];
+    onSave: (selectedFields: string[]) => void;
+};
+
+export const FieldSelectorModal: FC<FieldSelectorProps> = ({
+    isOpen,
+    onClose,
+    metadata,
+    initialSelectedFields,
+    onSave,
+}) => {
+    const getInitialSelectedFields = () => {
+        const fields = new Set(initialSelectedFields ?? getDefaultSelectedFields(metadata));
+        fields.add(ACCESSION_VERSION_FIELD);
+        return fields;
+    };
+
+    const [selectedFields, setSelectedFields] = useState<Set<string>>(getInitialSelectedFields());
+
+    const handleToggleField = (fieldName: string) => {
+        if (fieldName === ACCESSION_VERSION_FIELD) {
+            return;
+        }
+
+        const newSelectedFields = new Set(selectedFields);
+        if (newSelectedFields.has(fieldName)) {
+            newSelectedFields.delete(fieldName);
+        } else {
+            newSelectedFields.add(fieldName);
+        }
+        newSelectedFields.add(ACCESSION_VERSION_FIELD);
+
+        setSelectedFields(newSelectedFields);
+        onSave(Array.from(newSelectedFields));
+    };
+
+    const handleSelectAll = () => {
+        const newSelectedFields = new Set<string>();
+        metadata.forEach((field) => {
+            newSelectedFields.add(field.name);
+        });
+        setSelectedFields(newSelectedFields);
+        onSave(Array.from(newSelectedFields));
+    };
+
+    const handleSelectNone = () => {
+        const newSelectedFields = new Set<string>();
+        newSelectedFields.add(ACCESSION_VERSION_FIELD);
+        setSelectedFields(newSelectedFields);
+        onSave(Array.from(newSelectedFields));
+    };
+
+    // Group fields by header
+    const fieldsByHeader = metadata.reduce<Record<string, Metadata[]>>((acc, field) => {
+        const header = field.header ?? 'Other';
+        acc[header] = acc[header] ?? [];
+        acc[header].push(field);
+        return acc;
+    }, {});
+
+    // Sort headers alphabetically, but keep "Other" at the end
+    const sortedHeaders = Object.keys(fieldsByHeader).sort((a, b) => {
+        if (a === 'Other') return 1;
+        if (b === 'Other') return -1;
+        return a.localeCompare(b);
+    });
+
+    return (
+        <BaseDialog title='Select Fields to Download' isOpen={isOpen} onClose={onClose} fullWidth={false}>
+            <div className='min-w-[950px]'></div>
```

Comment:
Hmm, this looks like it an unintentional left-over.

Reply:
I think this was actually intentional but let's see its effect after https://github.com/loculus-project/loculus/pull/3869 which I think may have changed the underlying situation

Reply:
The purpose of this was to ensure this fully obscures the modal below to improve aesthetics. Other changes mean it now shorter than the modal below, which I'd prefer to tackle later if at all, but I think we can keep this to ensure this overlying modal is wider.